phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -2621,6 +2625,13 @@ private MetaDataMutationResult doDropTable(byte[] key, byte[] tenantId, byte[] s
                 List<ImmutableBytesPtr> invalidateList = new ArrayList<ImmutableBytesPtr>();
                 invalidateList.add(cacheKey);
                 PTable table = getTableFromCache(cacheKey, clientTimeStamp, clientVersion);
+
+                // Disallow if the parent has PHOENIX_TTL set.
+                if (parentTable != null &&  parentTable.getPhoenixTTL() != PHOENIX_TTL_NOT_DEFINED) {
```

Comment:
just want to confirm that we don't support more than 2 level case. For example, we have a global view V1 of the TTL set. The V2 is a view on top of the V1. If we create a new V3 on of the v2 with a TTL value, this won't catch and allow the V3 view to have a TTL value, which might conflict with V1 TTL.

Reply:
@yanxinyi  Currently the PHOENIX_TTL value is stored with every view and at every level - so in this case V2 would have the same PHOENIX_TTL value as of V1. So when creating V3 the parentTable.getPhoenixTTL() != PHOENIX_TTL_NOT_DEFINED. 

But realized as was going thru this that I need an additional check to ensure that the PHOENIX_TTL is being altered